bibtex-actions.el

```diff
@@ -234,12 +234,55 @@ If FORCE-REBUILD-CACHE is t, force reloading the cache."
     (bibtex-actions-refresh force-rebuild-cache))
   bibtex-actions--candidates-cache)
 
+(defun bibtex-actions-complete-key-at-point ()
+  "Complete citation key at point.
+
+When inserting '@' in a buffer the capf UI will present user with
+a list of entries, from which they can narrow against title,
+author, etc., and then select one.  This function will then return
+the key 'key', resulting in '@key' at point."
+
+  ;; Activate when user enters '@'.
+  ;;
+  ;; Notes:
+  ;;
+  ;;   - pandoc syntax is [@key], or can also be just @key.
+  ;;   - org syntax is stricter; [cite:see @key1 p23]
+  ;;   - latex uses keys, but sans '@' (I don't want to worry about this ATM though)
+  ;;
+  ;; Maybe this should be constrained further for org?
+  ;;
+  ;; Or, given that pandoc syntax can be used in org context, maybe condition
+  ;; this on 'bibtex-completion-format-citation-functions'?
+  (when (looking-back "@[a-zA-Z0-9]*" 5)
+    (let ((candidates (bibtex-actions--get-candidates)))
+      (list
+       (save-excursion
+         (backward-word)
+         (point))
+       (point)
+       (mapcar #'cdr candidates) :exclusive 'no))))
+
+(defun bibtex-actions-capf ()
+  "My test capf; alt."
+  (when (looking-back "@[a-zA-Z0-9]*" 5)
+    (let* ((candidates (bibtex-actions--get-candidates))
+           (begin (save-excursion
+                    (backward-word)
+                    (point)))
+           (end (point)))
+           (cdr
```

Comment:
This is my attempt to grab the cdr from the chosen candidate, but it doesn't work (I get a "no match").